mora/service/employee.py

```diff
@@ -442,29 +443,61 @@ def create_employee_relation(employee_uuid):
     }
 
     reqs = flask.request.get_json()
+    c = common.get_connector()
+
+    functions = []
+    rels = collections.defaultdict(list)
+    types = []
+
     for req in reqs:
         role_type = req.get('type')
         handler = handlers.get(role_type)
 
+        types.append(common.RELATION_TRANSLATIONS[role_type])
+
         if not handler:
             raise exceptions.HTTPException(
                 exceptions.ErrorCodes.E_UNKNOWN_ROLE_TYPE,
                 message=role_type)
 
         elif issubclass(handler, common.AbstractRelationDetail):
-            handler(common.get_connector().bruger).create(
-                str(employee_uuid),
-                req,
-            )
+            relname, rel = handler.get_relation_for(req)
+
+            rels[relname].append(rel)
 
         else:
-            handler(str(employee_uuid), req)
+            functions.append(handler(str(employee_uuid), req))
+
+    note = "Opret {}".format(', '.join(types))
 
+    if not rels:
         # Write a noop entry to the user, to be used for the history
         common.add_bruger_history_entry(
             employee_uuid,
-            "Opret {}".format(common.RELATION_TRANSLATIONS[role_type])
+            note,
         )
+    else:
```

Comment:
This kind of goes against the original intent of this function.

The idea was to have a relatively generic function, with a number of handlers sharing the same interface, keeping all of the specific nitty-gritty of creating the objects and calling LoRa in the specific handlers, rather than here.

I'd rather we modify the handlers for this to basically implement two functions -- one two create a payload, and one to send the payload to LoRa, and try to unify the two different types of handlers we have created here. That way, we still perform all the necessary validation for every object before updating the database, while keeping this function relatively simple.

Handling history in a elegant way _is_ a bit difficult, though, I realize.